### PR TITLE
explicitly link with hipRTC

### PR DESCRIPTION
**Pull request by JehandadKhan**
hipRTC symbols have been depracated from `libamdhip64.so.5` and now live in `libhiprtc.so`. This PR does a `find_package` for `hipRTC` to discover the location of the library file and then links it to the `MIOpen` target. The updated build system produces the correct linkage as shown by the output below:

``` bash
root@hpe-sjc2-43:/data/repos/MIOpen/build# ldd lib/libMIOpen.so
        linux-vdso.so.1 (0x00007ffc2199a000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007ffb3c894000)
        libhiprtc.so.5 => /opt/rocm/lib/libhiprtc.so.5 (0x00007ffb3c6e3000)
        libamd_comgr.so.2 => /opt/rocm/lib/libamd_comgr.so.2 (0x00007ffb33894000)
        librocblas.so.3 => /opt/rocm/lib/librocblas.so.3 (0x00007ffb14038000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007ffb14015000)
        librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007ffb1400b000)
        libroctx64.so.4 => /opt/rocm/lib/libroctx64.so.4 (0x00007ffb14004000)
        libamdhip64.so.5 => /opt/rocm/lib/libamdhip64.so.5 (0x00007ffb1253e000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007ffb1235c000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007ffb1220d000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007ffb121f2000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ffb12000000)
        /lib64/ld-linux-x86-64.so.2 (0x00007ffb7aeb2000)
        libhsa-runtime64.so.1 => /opt/rocm/lib/libhsa-runtime64.so.1 (0x00007ffb11d4a000)
        libnuma.so.1 => /lib/x86_64-linux-gnu/libnuma.so.1 (0x00007ffb11d3d000)
        libz.so.1 => /opt/rocm/lib/libz.so.1 (0x00007ffb11d21000)
        libtinfo.so.6 => /lib/x86_64-linux-gnu/libtinfo.so.6 (0x00007ffb11cf1000)
        libelf.so.1 => /lib/x86_64-linux-gnu/libelf.so.1 (0x00007ffb11cd5000)
        libdrm.so.2 => /opt/amdgpu/lib/x86_64-linux-gnu/libdrm.so.2 (0x00007ffb11cba000)
        libdrm_amdgpu.so.1 => /opt/amdgpu/lib/x86_64-linux-gnu/libdrm_amdgpu.so.1 (0x00007ffb11cac000)
```

### CODE REVIEW

**Comment by JehandadKhan** [comment]
cc @atamazov

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -759,6 +759,7 @@ if( MIOPEN_BACKEND STREQUAL "OpenCL")
     target_link_libraries( MIOpen PUBLIC ${OPENCL_LIBRARIES} )
     list(APPEND PACKAGE_DEPENDS PACKAGE OpenCL)
 elseif(MIOPEN_BACKEND STREQUAL "HIPOC" OR MIOPEN_BACKEND STREQUAL "HIP")
+    target_link_libraries(MIOpen PRIVATE hiprtc)
```
Comment (junliume) [inline src/CMakeLists.txt]: Not a bit deal but can we follow the format of the other similar lines?

Comment (JehandadKhan) [reply src/CMakeLists.txt]: Done

File: CMakeLists.txt [hunk 2]
```diff
@@ -206,6 +206,7 @@ endif()
 
 
 # HIP is always required
+find_package(hiprtc REQUIRED PATHS /opt/rocm)
```
Comment (atamazov) [inline CMakeLists.txt]: Please move this after line 256 and enclose into `if(MIOPEN_USE_HIPRTC) ... endif()`

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -759,6 +759,7 @@ if( MIOPEN_BACKEND STREQUAL "OpenCL")
     target_link_libraries( MIOpen PUBLIC ${OPENCL_LIBRARIES} )
     list(APPEND PACKAGE_DEPENDS PACKAGE OpenCL)
 elseif(MIOPEN_BACKEND STREQUAL "HIPOC" OR MIOPEN_BACKEND STREQUAL "HIP")
+    target_link_libraries( MIOpen PRIVATE hiprtc)
```
Comment (atamazov) [inline src/CMakeLists.txt]: Enclose into `if(MIOPEN_USE_HIPRTC) ... endif()`

**Review by atamazov — CHANGES_REQUESTED** [review]
Consider MIOPEN_USE_HIPRTC=Off

File: CMakeLists.txt (hunk 2 above)
Comment (junliume) [reply CMakeLists.txt]: Done

File: src/CMakeLists.txt (hunk 3 above)
Comment (junliume) [reply src/CMakeLists.txt]: Done

File: CMakeLists.txt (hunk 2 above)
Comment (JehandadKhan) [reply CMakeLists.txt]: Good Catch!

File: src/CMakeLists.txt (hunk 3 above)
Comment (atamazov) [reply src/CMakeLists.txt]: @JehandadKhan @junliume Hmm... seems that not. 

And this `target_link_libraries` for hiprtc already exists but only for WIN32:
https://github.com/ROCm/MIOpen/blob/4033d2191107f8381c4022612ca896019997429a/src/CMakeLists.txt#L765-L767

Maybe we should just remove `AND WIN32`.

Please also notice that this PR links `hiprtc` but WIN32 code links `hiprtc::hiprtc`.

File: CMakeLists.txt (hunk 2 above)
Comment (atamazov) [reply CMakeLists.txt]: [resolved]

File: src/CMakeLists.txt (hunk 3 above)
Comment (junliume) [reply src/CMakeLists.txt]: @atamazov  updated.

**Review by junliume — APPROVED** [review]

**Pull request merged**